datatypes/src/raster/helpers.rs

```diff
@@ -214,10 +214,35 @@ macro_rules! generate_generic_raster2d {
     };
 }
 
+/// Calls a function on a `TypedRaster2D` and some `RasterDataType`-like enum, effectively matching
+/// the raster with the corresponding enum value of the other enum.
+/// Call via `call_generic_raster2d_ext!(input, (raster, e) => function)`.
+#[macro_export]
+macro_rules! call_generic_raster2d_ext {
```

Comment:
will this scale up to more then 3 inputs?

Reply:
unsure but not the focus of this PR

Reply:
then we will discuss this as a different issue.